aws-cloudformation-schema/aws-dms-dataprovider.json

```diff
@@ -75,131 +75,126 @@
+          "additionalProperties" : false
+        },
+        "OracleSettings" : {
+          "description" : "OracleSettings property identifier.",
+          "type" : "object",
+          "properties" : {
+            "ServerName" : {
+              "type" : "string"
+            },
+            "Port" : {
+              "type" : "integer"
+            },
+            "DatabaseName" : {
+              "type" : "string"
+            },
+            "SslMode" : {
+              "type" : "object",
+              "$ref" : "#/definitions/DmsSslModeValue"
+            },
+            "CertificateArn" : {
+              "type" : "string"
+            },
+            "AsmServer" : {
+              "type" : "string"
+            },
+            "SecretsManagerOracleAsmSecretId" : {
+              "type" : "string"
+            },
+            "SecretsManagerOracleAsmAccessRoleArn" : {
+              "type" : "string"
+            },
+            "SecretsManagerSecurityDbEncryptionSecretId" : {
+              "type" : "string"
+            },
+            "SecretsManagerSecurityDbEncryptionAccessRoleArn" : {
+              "type" : "string"
             }
-          }
-        }
-      }, {
-        "description" : "MicrosoftSqlServerSettings property identifier.",
-        "type" : "object",
-        "additionalProperties" : false,
-        "properties" : {
-          "MicrosoftSqlServerSettings" : {
-            "type" : "object",
-            "additionalProperties" : false,
-            "properties" : {
-              "ServerName" : {
-                "type" : "string"
-              },
-              "Port" : {
-                "type" : "integer"
-              },
-              "DatabaseName" : {
-                "type" : "string"
-              },
-              "SslMode" : {
-                "type" : "object",
-                "$ref" : "#/definitions/DmsSslModeValue"
-              },
-              "CertificateArn" : {
-                "type" : "string"
-              }
+          },
+          "required" : [ "ServerName", "Port", "SslMode", "DatabaseName" ],
+          "additionalProperties" : false
+        },
+        "MicrosoftSqlServerSettings" : {
+          "description" : "MicrosoftSqlServerSettings property identifier.",
+          "type" : "object",
+          "properties" : {
+            "ServerName" : {
+              "type" : "string"
+            },
+            "Port" : {
+              "type" : "integer"
+            },
+            "DatabaseName" : {
+              "type" : "string"
+            },
+            "SslMode" : {
+              "type" : "object",
+              "$ref" : "#/definitions/DmsSslModeValue"
+            },
+            "CertificateArn" : {
+              "type" : "string"
             }
-          }
+          },
+          "required" : [ "ServerName", "Port", "SslMode", "DatabaseName" ],
+          "additionalProperties" : false
         }
-      } ]
+      },
+      "anyOf" : [ {
```

Comment:
It looks like this change is being viewed as a breaking change. 

```json
"type": "object",
"anyOf": [
    { 
        "type": "object",
        "properties": { "PostgresSqlSettings: {...} }
    }
]
```

to 

```json
"type": "object",
 "properties": { 
        "type": "object",
        "properties": { "PostgresSqlSettings: {...} }
 }
"anyOf": [{
   "required": [ "PostgresSqlSettings" ]
}]
```

Which changes the generation from

```ts

export interface Settings0PropertiesArgs {
        postgreSqlSettings?: pulumi.Input<inputs.dms.Settings0PropertiesPostgreSqlSettingsPropertiesArgs>;
}

export interface Settings1PropertiesArgs {
  mySqlSettings?: pulumi.Input<inputs.dms.Settings1PropertiesMySqlSettingsPropertiesArgs>;
}
```

to 

```ts
    export interface SettingsPropertiesArgs {
        /**
         * PostgreSqlSettings property identifier.
         */
        postgreSqlSettings?: pulumi.Input<inputs.dms.SettingsPropertiesPostgreSqlSettingsPropertiesArgs>;
        /**
         * MySqlSettings property identifier.
         */
        mySqlSettings?: pulumi.Input<inputs.dms.SettingsPropertiesMySqlSettingsPropertiesArgs>;
}
```

@t0yv0 I think  this is a no-op, what do you think?

Reply:
Hi thanks for bringing this up! Actually I'm not sure I'm being slightly intellectually lazy here, as reasoning through which change is breaking and which change isn't is specific to the language in Pulumi. I would love to have this reasoning automated with test cases verifying that the reasoning is "correct" - I think this can fit easily into the schema-tools purview. I'm going to check up if there's any existing tickets of this sort in there and if not add some.

Reply:
Jotted down in https://github.com/pulumi/schema-tools/issues/70

Reply:
I agree that we should improve the analysis done by schematools. 

For this provider, which is still in preview and is explicitly pitched as reflecting the CC api directly, I don't think our goal is even to avoid taking breaking changes from upstream. As long as we're faithfully reflecting the upstream, I'm in favor of shipping even breaking changes, at least until the provider is out of preview.

Reply:
(For this particular change, I think it's not breaking in TS and python, but probably would be a small break in go, java and dotnet since these languages require the user to use a named struct)

Reply:
> For this provider, which is still in preview and is explicitly pitched as reflecting the CC api directly, I don't think our goal is even to avoid taking breaking changes from upstream. As long as we're faithfully reflecting the upstream, I'm in favor of shipping even breaking changes, at least until the provider is out of preview.

Sounds good to me!